### PR TITLE
Project Survey

### PR DESCRIPTION
**Pull request by JHarde**
Link to my deployed project:

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/Form.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import Question1 from "./Question1";
+import Question2 from "./Question2";
+import Question3 from "./Question3";
+import Question4 from "./Question4";
+import Summary from "./Summary";
```
Comment (luuf81) [inline code/src/Form.js:6]: separate external/internal imports next time :)

File: code/src/Form.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+const Form = () => {
+  const [name, setName] = useState("");
+  const [city, setCity] = useState("");
+  const [training, setTraining] = useState([]);
+  const [experience, setExperience] = useState();
+  const [question, setQuestion] = useState("Question1");
```
Comment (luuf81) [inline code/src/Form.js:13]: Good clear naming of state and functions 👍

File: code/src/Form.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  const handleCityChange = (event) => {
+    setCity(event.target.value);
+  };
+
+  const handleTrainingChange = (trainingValue) => {
+    training.includes(trainingValue)
+      ? setTraining(training.filter((item) => item !== trainingValue))
+      : setTraining([...training, trainingValue]);
+  };
```
Comment (luuf81) [inline code/src/Form.js:39]: maybe comment on this function since it's not obvious to me what it does.

File: code/src/Helper.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const validateEmail = (email) => {
+  const isValidLength = email.length > 6;
+  const isValidEmail = /\S+@\S+\.\S+/.test(email);
```
Comment (luuf81) [inline code/src/Helper.js:3]: Regex validation 😍

File: code/src/Form.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  const handleNextQuestion = (event) => {
+    event.preventDefault();
+    const nextQuestion = questionList.indexOf(question);
+    setQuestion(questionList[nextQuestion + 1]);
+  };
+
+  return (
+    <>
+      {question !== "Summary" ? (
```
Comment (luuf81) [inline code/src/Form.js:49]: Great use of ternary operator here but maybe keep it consistent and do it like you do on the other ones? "&&"

File: code/src/Form.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    setCity(event.target.value);
+  };
+
+  const handleTrainingChange = (trainingValue) => {
+    training.includes(trainingValue)
+      ? setTraining(training.filter((item) => item !== trainingValue))
+      : setTraining([...training, trainingValue]);
+  };
+
+  const handleNextQuestion = (event) => {
```
Comment (luuf81) [inline code/src/Form.js:41]: no need for parenthesis when single argument

File: code/src/Form.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    setExperience(newValue);
+  };
+
+  const handleNameChange = (event) => {
+    setName(event.target.value);
+  };
+
+  const handleCityChange = (event) => {
+    setCity(event.target.value);
+  };
```
Comment (luuf81) [inline code/src/Form.js:33]: no need for curly brackets when only one line of code

File: code/src/Question1.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+const Question1 = ({ city, onCityChange }) => {
```
Comment (luuf81) [inline code/src/Question1.js:3]: Destructuring props 🙌

File: code/src/Question3.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+//               type="radio"
+//               name="radio"
+//               value={style}
+//               onChange={onTrainingChange}
+//               checked={training === styles}
+//               id={style}
+//               className="radio"
+//               required
+//               role="radio"
+//               aria-labelledby="chocie a type of training"
```
Comment (luuf81) [inline code/src/Question3.js:60]: Commented out code should be deleted :)

File: code/src/index.css [hunk 10]
```diff
@@ -5,9 +5,121 @@ body {
...
+  font-size: 30px;
+}
+
+/* .start-text {
+  display: flex;
+  justify-content: center;
+  font-family: 'Amatic SC', cursive;
+  color: white;
+  font-size: 25px;
+} */
```
Comment (luuf81) [inline code/src/index.css:38]: commented out => delete

File: code/src/index.css [hunk 11]
```diff
@@ -5,9 +5,121 @@ body {
...
+
+h2 {
+  display: flex;
+  justify-content: center;
+}
+
+h3 {
+  display: flex;
+  justify-content: center;
+}
```
Comment (luuf81) [inline code/src/index.css:48]: doing the same thing right? h2, h3 {}...

File: code/src/index.css [hunk 12]
```diff
@@ -5,9 +5,121 @@ body {
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  
+}
+
+.submit-button {
+  /* display: flex;
+  justify-content: center;
+  align-items: center; */
```
Comment (luuf81) [inline code/src/index.css:95]: delete commented out code

File: code/src/Question2.js [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+      <Slider
+        defaultValue={20}
+        valueLabelFormat={valueLabelFormat}
+        getAriaValueText={valuetext}
+        aria-labelledby="discrete-slider-restrict"
+        step={10}
+        onChange={handleSliderChange}
+        valueLabelDisplay="auto"
+        marks={marks}
+      />
```
Comment (luuf81) [inline code/src/Question2.js:67]: This slider doesn´t seem to work correctly so either remove or fix? 😼 Love the effort!

**Review by luuf81 — APPROVED** [review]
Great work Joel! Nice and clean design and love the fact you tried something new with Material UI! 

Slider is red level requirement but still maybe something to fix?

File: code/src/Question3.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+  return (
+    <>
+      <section>
+        <div className="question-3">
+          <div>
+            <label htmlFor="trainingKickboxing">Kickboxing</label>
+            <input
+              id="trainingKickboxing"
+              type="radio"
+              checked={training.includes("Kickboxing")}
```
Comment (HIPPIEKICK) [inline code/src/Question3.js:13]: This is how you would deal with checkboxes, not radio buttons :) Radio buttons work like only one at a time should be selected.

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Great effort but some things need to be fixed:
- Radio buttons should act like radio buttons
- Range slider is not working properly (every other output shows 0 and no state is changing when the value is updated)
- Clean up your code (remove unused code)

**Comment by JHarde** [comment]
Have done my best to implement the requested changes now, let me know if more is needed!

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**